File: backend/test_wa.py
import requests
import json
import sys

# User's provided config
URL = "http://157.151.161.62:8080"
INSTANCE = "suporte_profamilia"
API_KEY = "casauniao_secret_key"

def test_send():
    endpoint = f"{URL}/message/sendText/{INSTANCE}"
    
    # We will test with a dummy number first just to see if it authenticates 
    # and accepts the payload format, or we can use the user's number if we knew it.
    # Let's use a standard dummy Brazilian mobile number.
    payload = {
        "number": "5511999999999",
        "text": "Teste de integração automática"
    }
    
    headers = {
        "apikey": API_KEY,
        "Content-Type": "application/json"
    }
    
    print(f"Sending to {endpoint}...")
    try:
        response = requests.post(endpoint, json=payload, headers=headers)
        print(f"Status Code: {response.status_code}")
        print("Response:", response.text)
    except Exception as e:
        print("Error:", e)

if __name__ == "__main__":
    test_send()

File: backend/test_wa.js
const axios = require('axios');

const URL = "http://157.151.161.62:8080";
const INSTANCE = "suporte_profamilia";
const API_KEY = "casauniao_secret_key";

async function testSend() {
    const endpoint = `${URL}/message/sendText/${INSTANCE}`;
    const payload = {
        number: "5511999999999",
        text: "Teste de integração da plataforma Profamília"
    };

    console.log(`Sending to ${endpoint}...`);
    try {
        const res = await axios.post(endpoint, payload, {
            headers: {
                apikey: API_KEY,
                'Content-Type': 'application/json'
            }
        });
        console.log("Status:", res.status);
        console.log("Response:", res.data);
    } catch (e) {
        console.error("Error Status:", e.response?.status);
        console.error("Error Data:", JSON.stringify(e.response?.data, null, 2) || e.message);
    }
}

testSend();
